### PR TITLE
feat(console): add connector

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add connector selection modal, this is used for adding or replacing connector.

## Linear Issue Reference

LOG-1725

## Testing

截屏2022-03-21 下午5 03 22

截屏2022-03-21 下午5 03 30

### CODE REVIEW

File: packages/console/src/pages/ConnectorDetails/index.tsx [hunk 1]
```diff
@@ -112,17 +117,40 @@ const ConnectorDetails = () => {
...
               {data.metadata.type === ConnectorType.Email && (
-                <ActionMenuItem>{t('connector_details.options_change_email')}</ActionMenuItem>
+                <ActionMenuItem
+                  icon={<Reset />}
+                  onClick={() => {
+                    setIsSetupOpen(true);
+                  }}
+                >
+                  {t('connector_details.options_change_email')}
+                </ActionMenuItem>
```
Comment (simeng-li) [inline packages/console/src/pages/ConnectorDetails/index.tsx:141]: the only difference is text ?

File: packages/console/src/pages/Connectors/components/SetupModal.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    if (type === ConnectorType.Email) {
+      return 'connectors.setup_title.email';
+    }
+
+    if (type === ConnectorType.SMS) {
+      return 'connectors.setup_title.sms';
+    }
+
+    return 'connectors.setup_title.social';
+  }, [type]);
```
Comment (simeng-li) [inline packages/console/src/pages/Connectors/components/SetupModal.tsx:46]: nit: as we have multiple places need to render different content based on different connector types, maybe we could have a static config some ware like. `DummyConnectorConfig[type].cardTitle`, `DummyConnectorConfig[type].details`

**Review by xiaoyijun — CHANGES_REQUESTED** [review]

File: packages/console/src/pages/ConnectorDetails/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/ConnectorDetails/index.tsx:141]: updated, PTAL

File: packages/console/src/pages/Connectors/components/SetupModal.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/Connectors/components/SetupModal.tsx:46]: Good call, but we only have `cardTitle` for now.

**Review by xiaoyijun — APPROVED** [review]

File: packages/console/src/pages/Connectors/components/SetupModal.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    [data, type]
+  );
+
+  return (
+    <ReactModal
+      isOpen={isOpen}
+      className={modalStyles.content}
+      overlayClassName={modalStyles.overlay}
+    >
+      <Card className={styles.card}>
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/console/src/pages/Connectors/components/SetupModal.tsx]: updated

**Comment by wangsijie** [comment]
@xiaoyijun add some changes to `<ModalLayout />`, PTAL

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**